dex/farm/src/rewards.rs

```diff
@@ -152,8 +152,12 @@ pub trait RewardsModule:
     }
 
     #[endpoint]
-    fn end_produce_rewards(&self) -> SCResult<()> {
+    fn end_produce_rewards_as_owner(&self) -> SCResult<()> {
```

Comment:
`#[only_owner]` here as well? :)

Reply:
no need - as at the next line you have self.require_permissions. code is good.

Reply:
I prefer either all only_owner or all require perms and I wouldnt change all to only_owner since its the v2 SCs and will soon be unused